library/spdm_secured_message_lib/libspdm_secmes_context_data.c

```diff
@@ -186,30 +186,22 @@ bool libspdm_secured_message_import_dhe_secret(void *spdm_secured_message_contex
     return true;
 }
 
-/**
- * Export the export_master_secret from an SPDM secured message context.
- *
- * @param  spdm_secured_message_context    A pointer to the SPDM secured message context.
- * @param  export_master_secret           Indicate the buffer to store the export_master_secret.
- * @param  export_master_secret_size       The size in bytes of the export_master_secret.
- *
- * @retval RETURN_SUCCESS  export_master_secret is exported.
- */
 bool libspdm_secured_message_export_master_secret(
     void *spdm_secured_message_context, void *export_master_secret,
     size_t *export_master_secret_size)
 {
     libspdm_secured_message_context_t *secured_message_context;
 
     secured_message_context = spdm_secured_message_context;
-    if (*export_master_secret_size < secured_message_context->hash_size) {
+
+    if (*export_master_secret_size > secured_message_context->hash_size) {
         *export_master_secret_size = secured_message_context->hash_size;
-        return false;
     }
+
     libspdm_copy_mem(export_master_secret, *export_master_secret_size,
```

Comment:
have you tested this?

```
 * Asserts and returns a non-zero value if any of the following are true:
 *   3) "src_len" is greater than "dst_len".
```
https://github.com/jyao1/libspdm/blob/readme_build/include/hal/library/memlib.h#L26

Reply:
Heh I did not. That's what I get for trying to rush this through before the end of the workday. I'll fix it and make sure it's tested.